conans/test/functional/toolchains/apple/test_xcodebuild_targets.py

```diff
@@ -492,13 +129,25 @@ def build(self):
                 xcode.build("HelloLibrary.xcodeproj", target="hello-static")
 
         def package(self):
-            copy(self, "*/libhello*", src=self.build_folder, dst=os.path.join(self.package_folder, "lib"), keep_path=False)
+            name = "hello-dynamic.dylib" if self.options.shared else "libhello-static.a"
+            copy(self, "build/{}/{}".format(self.settings.build_type, name),
+                 src=self.build_folder, dst=os.path.join(self.package_folder, "lib"), keep_path=False)
             copy(self, "*/*.hpp", src=self.build_folder, dst=os.path.join(self.package_folder, "include"), keep_path=False)
 
         def package_info(self):
-            self.cpp_info.libs = ["hello"]
+            self.cpp_info.libs = ["hello-{}".format("dynamic.dylib" if self.options.shared else "static")]
     """)
 
+static_xcconfig = textwrap.dedent("""
```

Comment:
I had to use these wrappers because it was not possible to set `LD_DYLIB_INSTALL_NAME` in the project.yml, xcodegen would not add this to the project